src/amo/reducers/home.js

```diff
@@ -249,11 +252,28 @@ type Action =
   | LoadHomeDataAction
   | SetClientAppAction;
 
-const createInternalAddons = (
-  response: ApiAddonsResponse,
+export const createInternalShelf = (
+  result: ExternalResultShelfType,
   lang: string,
-): Array<AddonType> => {
-  return response.results.map((addon) => createInternalAddon(addon, lang));
+): ResultShelfType => {
+  const sliceEnd =
+    ADDON_TYPE_STATIC_THEME === result.addons[0].type
+      ? LANDING_PAGE_THEME_COUNT
+      : LANDING_PAGE_EXTENSION_COUNT;
+
+  const shelfAddons = result.addons.slice(0, sliceEnd).map((addon) => {
+    return createInternalAddon(addon, lang);
+  });
+
+  return {
+    title: result.title,
```

Comment:
Here's one of the spots where you'll need to deal with the fact that the API is now returning localized strings. There are others as well.